### DISCUSSION TITLE
Colors for boolean values and negative numbers

### DISCUSSION THREAD

**Discussion by vanillajonathan** [minhhungit/ConsoleTableExt#14] [Ideas]
The boolean value **true** could be displayed in green color while the boolean value **false** could be displayed in red color.

Negative numbers (such as `−123`) could be displayed in red color.

**Comment by minhhungit** [comment]
- I think it is better we should have something like Formatter which help user define column formatter by themeself, input is original text and output is formatted text, for example:

```csharp
builder.WithFormatter(col_index, (text) => $"Hello {text}");
```
I will implement it soon

- About color of text , it is quite complex to implement since table builder return a `StringBuilder`, then we just print all it using `Console.WriteLine(strBuilder)`, it is quite hard to change color for some characters, it also relate to performance

If ignore perfomance we can return more flexible type. It's tradeoff.

If you have any better idea, please let me know, and fell free to create Pull request

**Reply by vanillajonathan** [reply]
User-definable column formatters sounds like a great idea!

**Reply by minhhungit** [reply]
@vanillajonathan version **3.1.2** is released that support WithFormatter

Here is a sample how to use:

```csharp
ConsoleTableBuilder.From(SampleListData)
    .WithTextAlignment(new Dictionary<int, TextAligntment> {
        { 1, TextAligntment.Right },
        { 3, TextAligntment.Right },
        { 100, TextAligntment.Right }
    })
    .WithMinLength(new Dictionary<int, int> {
        { 1, 30 }
    })
    .WithCharMapDefinition(CharMapDefinition.FramePipDefinition)
    .WithTitle("HELLO I AM TITLE", ConsoleColor.Green, ConsoleColor.DarkGray)
    .WithFormatter(1, (text)=> {
        return text.ToUpper().Replace(" ", "-") + " «";
    })                
    .ExportAndWriteLine(TableAligntment.Center);
```

It should print out result like this:

![image](https://user-images.githubusercontent.com/2279508/107730975-53157780-6d27-11eb-83c6-68f51ed0c055.png)


Hope it helps
